optimizations.qmd

```diff
@@ -1,90 +1,860 @@
+
+![](images/efficientnumerics_alexnet.png)
+
+Src: [https://arxiv.org/abs/1510.00149](https://arxiv.org/abs/1510.00149)
+
+[Figure]()
+
+### Quantization and Pruning
+
+Pruning and quantization work well together, and it's been found that pruning doesn't hinder quantization. In fact, pruning can help reduce quantization error. Intuitively, this is due to pruning reducing the number of weights to quantize, thereby reducing the accumulated error from quantization. For example, an unpruned AlexNet has 60 million weights to quantize whereas a pruned AlexNet only has 6.7 million weights to quantize. This significant drop in weights helps reduce the error between quantizing the unpruned AlexNet vs. the pruned AlexNet. Furthermore, recent work has found that quantization-aware pruning generates more computationally efficient models than either pruning or quantization alone; It typically performs similar to or better in terms of computational efficiency compared to other neural architecture search techniques like Bayesian optimization [Ps and Qs: Quantization-Aware Pruning for Efficient Low Latency Neural Network Inference][2021](https://arxiv.org/pdf/2102.11289.pdf).
+
+![](images/efficientnumerics_qp1.png)
+
+Src: [https://arxiv.org/abs/1510.00149](https://arxiv.org/abs/1510.00149)
+
+![](images/efficientnumerics_qp2.png)
+
+### Edge-aware Quantization
+
+Quantization not only reduces model size but also enables faster computations and draws less power, making it vital to edge development. Edge devices typically have tight resource constraints with compute, memory, and power, which are impossible to meet for many of the deep NN models of today. Furthermore, edge processors do not support floating point operations, making integer quantization particularly important for chips like GAP-8, a RISC-=V SoC for edge inference with a dedicated CNN accelerator, which only support integer arithmetic..
+
+One hardware platform utilizing quantization is the ARM Cortex-M group of 32=bit RISC ARM processor cores. They leverage fixed-point quantization with power of two scaling factors so that quantization and dequantization can be efficiently done by bit shifting. Additionally, Google Edge TPUs, Google's emerging solution for running inference at the edge, is designed for small, low-powered devices and can only support 8-bit arithmetic. Recently, there has been significant strides in the computing power of edge processors, enabling the deployment and inference of costly NN models previously limited to servers.
```

Comment:
Is it supposed to be "RISC-V SoC" and "32-bit RISC ARM" instead of "RISC-=V SoC" and "32=bit RISC ARM"?